th2_data_services/interfaces/events_tree/events_tree_collection.py

```diff
@@ -46,12 +57,19 @@ def __init__(
             preserve_body: If True it will preserve 'body' field in the Events.
             stub: If True it will create stub when event is broken.
         """
+        self._id = id(self)
         self._preserve_body = preserve_body
         self._roots: List[EventsTree] = []
         self._parentless: Optional[List[EventsTree]] = None
         self._detached_nodes: Dict[Optional[str], List[Node]] = defaultdict(list)  # {parent_event_id: [Node1, ..]}
         self._stub_status = stub
         self._data_source = data_source
+        self._logger = _EventsTreeCollectionLogger(logger, {"id": self._id})
+        self._warnings = []
```

Comment:
Why do we need this warning list?

Reply:
Warning list needed for tests in test_events_tree_collection in order to understand whether these warnings are displayed or not. But I'm trying to use the catch_warning() function to catch and issue warnings in tests in test_events_tree_collection. After that I will get rid of the warnings list.

Reply:
So we are using this for testing purposes?

Reply:
Warnings list was removed from EventsTreeCollection class. Now the output of warnings is checked by a test in a tests_events_tree_collection.py (256): https://github.com/th2-net/th2-data-services/pull/131/commits/0460ed688d47f24a298ed465e68c81b9cfbf7118